integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala

```diff
@@ -836,8 +836,15 @@ object GlobalDictionaryUtil {
       }
     } catch {
       case ex: Exception =>
-        LOGGER.error(ex, "generate global dictionary failed")
-        throw ex
+        ex match {
+          case spx: SparkException =>
+            LOGGER.error(spx, "generate global dictionary failed")
+            throw new Exception(
+              "Global dictionary generation failed. Please check the logs.")
```

Comment:
We cannot generally handle this SparkException, also we have to avoid "check logs". So handle specific TextParseException and compose required error message, during parsing only.